### ISSUE TITLE
How to add ENT labels to ENE_ONTONOTES_MAPPING

### ISSUE THREAD

**Issue by hiroshi-matsuda-rit** [megagonlabs/ginza#140]
I would like to add some new labels to ginza, like ‘speech’ and ‘music’ and to classify these new entities. I think I need to add these new labels to this ‘ene_ontonotes_mapper.py’. But I am not sure how to retrain this ginza model to predict these labels?
https://github.com/megagonlabs/ginza/blob/develop/ginza/ene_ontonotes_mapper.py
https://github.com/megagonlabs/UD_Japanese-GSD/blob/master/ene_mapping.xlsx

**Comment by hiroshi-matsuda-rit** [comment]
@YiAthena Please use this issue.

**Comment by YiAthena** [comment]
Hi @hiroshi-matsuda-rit , thanks for opening the issue. I would like to add some new labels to ginza, like ‘speech’ and ‘music’ and to classify these new entities.  I think I need to add these new labels to this ‘ene_ontonotes_mapper.py’. But I am not sure how to retrain this ginza model to predict these labels?

**Comment by hiroshi-matsuda-rit** [comment]
Thank you for asking.

I'd like to describe the basics of GiNZA's NER first.
The NER model of GiNZA uses Sekine's Extended Named Entity (ENE) label system in the pipeline of spaCy.
The ENE label set is defined in https://github.com/megagonlabs/ginza/blob/develop/ginza/ene_ontonotes_mapper.py .
Also, you can get spaCy's OntoNotes-based NE labels via `spacy.ene_type_(Token)` or `spacy.ene_label_ontonotes(Token)`
The mapping from ENE to OntoNotes is defined in the document below.
https://github.com/megagonlabs/UD_Japanese-GSD/blob/master/ene_mapping.xlsx

Next, I'd describe how to extend the NE labels on GiNZA.
We can use `Language.update()` to the pipeline of GiNZA as well as spaCy.
https://spacy.io/usage/training#example-train-ner

**Comment by hiroshi-matsuda-rit** [comment]
@YiAthena I revised above comment. Please read it again.

**Comment by hiroshi-matsuda-rit** [comment]
@YiAthena I'd like to close this issue. Please feel free to reopen this issue or open new related issues.

**Comment by YiAthena** [comment]
Hi @hiroshi-matsuda-rit sorry for my late reply. I successfully added new NE to the Ginza model. Thank you very much for your kind help!

I have another small question. Sometimes there is a number attached to the printed entity label. For example:
print("Entities", [(ent.text. ent.label_)])
Entities[('13時', 'TIME40')]
I don't know what does '40' mean. I understand this might be a space's question. I searched but didn't  find the answer. 
I would be appreciate if I could get any information.
